### PR TITLE
Add AWS documentation

### PR DESCRIPTION
**Pull request by giovannt0**
Signed-off-by: Theo Giovanna <gtheo@google.com>

### CODE REVIEW

File: docs/usermanual/index.md [hunk 1]
```diff
@@ -17,16 +17,156 @@ TODO
...
+
+# Create a forensic copy of the boot volume of instance 'inst1' in the default account.
+
+copy = forensics.CreateVolumeCopy('us-east-2b',
+                                  instance_id='inst1')
+```
+
+#### Using the CLI
+
+All snippets of code above have their CLI equivalent:
```
Comment (sa3eed3ed) [inline docs/usermanual/index.md:91]: can we add each cli snippet below the python example ? it is easier to follow for the reader.

File: docs/usermanual/index.md [hunk 2]
```diff
@@ -17,16 +17,156 @@ TODO
...
+directly. Instead, you must first make a copy of the snapshot, and re-encrypt it with a Customer Managed Key (CMK) key. 
+Both the CMK key and the snapshot copy need to then be shared with the destination account.
+
+
+To reduce key management hassle, libcloudforensics allows you to transfer encrypted EBS resources between accounts by 
+generating a one-time use CMK key. This key is deleted once the process completes. The process is depicted below:
+
+![ebs-encryption](https://lh4.googleusercontent.com/U_DxaWaZqx00gSG8kfhlwB6dkGGcPi5_Cd_qgZR9VMdW0rWmrHD-VDeChMKNBfJJ86qBJC9duaZzHbmBxWU1WpsNkNYmp3UQvmU2R9EuDmn0vJJqy0rZVTjZVBePIXvb8QlJmBgq)
+
+All of the code snippets and command lines given in the previous section can be applied as-is, regardless of the target
```
Comment (sa3eed3ed) [inline docs/usermanual/index.md:121]: can you add relative link to previous section?

File: docs/usermanual/index.md [hunk 3]
```diff
@@ -17,16 +17,156 @@ TODO
...
+volume using EBS encryption or not.
 
 ### Analysis VMs
 
-TODO
+Use case: you have a volume `vol1` in an AWS account (Account A, in zone `us-east-2b`) which you would like to make a copy 
+of in a different account (Account B). Afterwards, you want to start an analysis VM in Account B, and attach the disk 
+copy that you created to start with your investigation.
+
+Your AWS credentials configuration should be similar to what is described above.
```
Comment (sa3eed3ed) [inline docs/usermanual/index.md:130]: also relative link to described above

File: docs/usermanual/index.md [hunk 4]
```diff
@@ -17,16 +17,156 @@ TODO
...
+`vm-forensics`) and click on `Connect`.
+
+``` important:: Pro tip: you can export an environment variable 'STARTUP_SCRIPT' that points to a custom bash script of
+    yours. This script will be shipped to the instance being created and executed during the first boot. You can 
+    therefore do any kind of pre-processing you want in this script.
+```
+
+#### Using the CLI 
+
+The volume copy CLI is identical to what is presented in previous sections.
```
Comment (sa3eed3ed) [inline docs/usermanual/index.md:165]: can we just add the cli cmd that copies the volume here before creating the analysis VM, this makes it consistent to the python example and clearer that the volume must be residing under the destination profile where you will start the VM.

**Comment by giovannt0** [comment]
Thanks @sa3eed3ed, updated

File: docs/usermanual/index.md [hunk 5]
```diff
@@ -17,16 +17,159 @@ TODO
...
+aws_access_key_id = yyy
+aws_secret_access_key = yyy
+
+# Access credentials to the default account to use with AWS operations. Note that this can be the same account as one 
+# of the accounts above. This is the account that will be used by libcloudforensics for AWSAccount objects that are not 
+# instantiated with a particular profile.
+[default]
+aws_access_key_id = zzz
+aws_secret_access_key = zzz
+```
```
Comment (tomchop) [inline docs/usermanual/index.md:44]: Maybe move this to a `prerequisites` or `setup` section or something? Also fine to leave for a future PR

File: docs/usermanual/index.md [hunk 6]
```diff
@@ -17,16 +17,159 @@ TODO
...
+Making copies of encrypted resources is a bit more involved. By default, encrypted Elastic Block Store (EBS) resources 
+use the aws/ebs key that is associated with the AWS account. Snapshots that are encrypted with this key cannot be shared 
+directly. Instead, you must first make a copy of the snapshot, and re-encrypt it with a Customer Managed Key (CMK) key. 
+Both the CMK key and the snapshot copy need to then be shared with the destination account.
+
+
+To reduce key management hassle, libcloudforensics allows you to transfer encrypted EBS resources between accounts by 
+generating a one-time use CMK key. This key is deleted once the process completes. The process is depicted below:
+
+![ebs-encryption](https://lh4.googleusercontent.com/U_DxaWaZqx00gSG8kfhlwB6dkGGcPi5_Cd_qgZR9VMdW0rWmrHD-VDeChMKNBfJJ86qBJC9duaZzHbmBxWU1WpsNkNYmp3UQvmU2R9EuDmn0vJJqy0rZVTjZVBePIXvb8QlJmBgq)
```
Comment (tomchop) [inline docs/usermanual/index.md:121]: I encourage you to host this image within this git repo

**Comment by tomchop** [comment]
It would be nice to have a native speaker (e.g. @Fryyyyy) review this as well.

File: docs/usermanual/index.md (hunk 5 above)
Comment (giovannt0) [reply docs/usermanual/index.md:44]: Sounds good, added section

File: docs/usermanual/index.md (hunk 6 above)
Comment (giovannt0) [reply docs/usermanual/index.md]: Added the image to the repo. Note that the image will be live only once this is merged to master.

File: docs/usermanual/index.md [hunk 7]
```diff
@@ -17,16 +17,161 @@ TODO
...
+of in a different account (Account B).
+
+#### Using the library and the CLI
+
+To make the copy, import the `forensics` package and use the `CreateVolumeCopy` method.
+
+```python
+from libcloudforensics.providers.aws import forensics
+
+# Create a forensic copy of the volume 'vol1' from one account to another
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:55]: Is there a particular difference between a copy and a 'forensic copy' in this context?

File: docs/usermanual/index.md [hunk 8]
```diff
@@ -17,16 +17,161 @@ TODO
...
+```bash
+# Create a forensic copy of the boot volume of instance 'inst1' in the default account
+libcloudforensics aws 'us-east-2b' copydisk --instance_id='inst1'
+```    
 
 #### EBS encryption
 
-TODO
+Making copies of encrypted resources is a bit more involved. By default, encrypted Elastic Block Store (EBS) resources 
+use the aws/ebs key that is associated with the AWS account. Snapshots that are encrypted with this key cannot be shared
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:115]: AWS/EBS if these are acronyms or `aws/ebs` if it's the technical name of the resource

File: docs/usermanual/index.md [hunk 9]
```diff
@@ -17,16 +17,161 @@ TODO
...
+# Create a forensic copy of the boot volume of instance 'inst1' in the default account
+libcloudforensics aws 'us-east-2b' copydisk --instance_id='inst1'
+```    
 
 #### EBS encryption
 
-TODO
+Making copies of encrypted resources is a bit more involved. By default, encrypted Elastic Block Store (EBS) resources 
+use the aws/ebs key that is associated with the AWS account. Snapshots that are encrypted with this key cannot be shared 
+directly. Instead, you must first make a copy of the snapshot, and re-encrypt it with a Customer Managed Key (CMK) key.
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:116]: don't need the last "key"

File: docs/usermanual/index.md [hunk 10]
```diff
@@ -17,16 +17,161 @@ TODO
...
+libcloudforensics aws 'us-east-2b' copydisk --instance_id='inst1'
+```    
 
 #### EBS encryption
 
-TODO
+Making copies of encrypted resources is a bit more involved. By default, encrypted Elastic Block Store (EBS) resources 
+use the aws/ebs key that is associated with the AWS account. Snapshots that are encrypted with this key cannot be shared 
+directly. Instead, you must first make a copy of the snapshot, and re-encrypt it with a Customer Managed Key (CMK) key. 
+Both the CMK key and the snapshot copy need to then be shared with the destination account.
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:117]: -key

File: docs/usermanual/index.md [hunk 11]
```diff
@@ -17,16 +17,161 @@ TODO
...
+
+
+To reduce key management hassle, libcloudforensics allows you to transfer encrypted EBS resources between accounts by 
+generating a one-time use CMK key. This key is deleted once the process completes. The process is depicted below:
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:121]: -key

File: docs/usermanual/index.md [hunk 12]
```diff
@@ -17,16 +17,161 @@ TODO
...
+
+![ebs-encryption](https://github.com/google/cloud-forensics-utils/tree/master/docs/source/images/ebs.png)
+
+All of the code snippets and command lines given in the [previous section](#using-the-library-and-the-cli) can be 
+applied as-is, regardless of the target volume using EBS encryption or not.
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:126]: regardless of whether the target...

File: docs/usermanual/index.md [hunk 13]
```diff
@@ -17,16 +17,161 @@ TODO
...
 
 ### Analysis VMs
 
-TODO
+Use case: you have a volume `vol1` in an AWS account (Account A, in zone `us-east-2b`) which you would like to make a copy 
+of in a different account (Account B). Afterwards, you want to start an analysis VM in Account B, and attach the disk 
+copy that you created to start with your investigation.
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:132]: s/start with/begin/

File: docs/usermanual/index.md [hunk 14]
```diff
@@ -17,16 +17,161 @@ TODO
...
+
+Your AWS credentials configuration should be similar to what is described [above](#aws).
+
+Using the library, you can start an analysis VM as follow:
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:136]: follows

File: docs/usermanual/index.md [hunk 15]
```diff
@@ -17,16 +17,161 @@ TODO
...
+# Start an analysis VM 'vm-forensics' for investigation in the AWS account 
+# dst_profile, and attach a volume to it.
+libcloudforensics aws 'us-east-2b' startvm 'vm-forensics' --boot_volume_size=50 --cpu_cores=4 --attach_volumes='vol1-copy' --dst_profile='dst_profile'
+```
+
+You're now ready to go! Log in your AWS account, find the instance (you can search it based on the name tag 
+`vm-forensics`) and click on `Connect`.
+
+``` important:: Pro tip: you can export an environment variable 'STARTUP_SCRIPT' that points to a custom bash script of
+    yours. This script will be shipped to the instance being created and executed during the first boot. You can
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:175]: -"of yours"

File: docs/usermanual/index.md [hunk 16]
```diff
@@ -17,16 +17,161 @@ TODO
...
+# dst_profile, and attach a volume to it.
+libcloudforensics aws 'us-east-2b' startvm 'vm-forensics' --boot_volume_size=50 --cpu_cores=4 --attach_volumes='vol1-copy' --dst_profile='dst_profile'
+```
+
+You're now ready to go! Log in your AWS account, find the instance (you can search it based on the name tag 
+`vm-forensics`) and click on `Connect`.
+
+``` important:: Pro tip: you can export an environment variable 'STARTUP_SCRIPT' that points to a custom bash script of
+    yours. This script will be shipped to the instance being created and executed during the first boot. You can 
+    therefore do any kind of pre-processing you want in this script.
```
Comment (Fryyyyy) [inline docs/usermanual/index.md:176]: -therefore

File: docs/usermanual/index.md (hunk 8 above)
Comment (giovannt0) [reply docs/usermanual/index.md:115]: It's the technical name (alias of the default AWS managed key)

File: docs/usermanual/index.md (hunk 7 above)
Comment (giovannt0) [reply docs/usermanual/index.md:55]: No, removed

**Comment by giovannt0** [comment]
Thanks @Fryyyyy, updated. @tomchop PTAL

**Pull request merged**